src/Matterhorn/Events.hs

```diff
@@ -156,8 +157,17 @@ formatError (NoSuchHelpTopic topic) =
     let knownTopics = ("  - " <>) <$> helpTopicName <$> helpTopics
     in "Unknown help topic: `" <> topic <> "`. " <>
        (T.unlines $ "Available topics are:" : knownTopics)
-formatError (BadAttachmentPath e msg) =
-    msg <> " The exception encountered was:\n " <> T.pack (displayException e)
+formatError (BadAttachmentPath e) =
+    case fromException e of
+      Just (ioe :: IO.IOError) -> 
+          if IO.isDoesNotExistError ioe
+          then "Error attaching, file does not exist!"
+          else if IO.isPermissionError ioe
+               then "Error attaching, lacking permissions to read file!"
+               else "Unable to attach the requested file.  Check that it exists and has proper permissions."
+      Nothing -> "Unknown error attaching file!\n" <>
+          "Please report this error at https://github.com/matterhorn-chat/matterhorn/issues"
```

Comment:
In this case, could you `mhLog` the value of `show e` so users can give us that when reporting?

Reply:
`formatError` is a pure function.  I can do this but it will necessarily involve a somewhat larger change.  Is that okay?

Reply:
To confirm here: no need to refactor this, please disregard my `mhLog` suggestion.